### ISSUE TITLE
Allow all gridmaps in scene to share same floor

### ISSUE THREAD

**Issue by elvisish** [godotengine/godot-proposals#6963]
### Describe the project you are working on

FPS game.

### Describe the problem or limitation you are having in your project

Currently if using multiple gridmaps in a scene, each gridmap will track it's own separate floor position, so if switching between gridmaps and trying to place different types of meshes on the same floor, the floor will have to be found separately for each gridmap (which can be incredible cumbersome if using several gridmaps and working in different floors frequently).

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow an option that syncs all gridmap floors with each other in the scene.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Gridmap core code that would allow the floors to be synchronized between all gridmaps in the scene.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, it would be gridmap core code.

### Is there a reason why this should be core and not an add-on in the asset library?

It would need to part of gridmap core code.

**Comment by Calinou** [comment]
I don't understand why you'd need them to be kept in sync. Can't you have multiple GridMap nodes at the same Y coordinate? The point of having multiple GridMap nodes is that you can place overlapping tiles.

**Comment by elvisish** [comment]
> I don't understand why you'd need them to be kept in sync. Can't you have multiple GridMap nodes at the same Y coordinate? The point of having multiple GridMap nodes is that you can place overlapping tiles.

They are all at exactly the same position, but changing floor is per-gridmap node:
![image](https://github.com/godotengine/godot-proposals/assets/16231628/69db8cc8-729d-4c3d-a2f3-1bb579f88efb)

> The point of having multiple GridMap nodes is that you can place overlapping tiles.

Exactly, that's why I have different gridmaps for different things, but if I'm moving around the scene and working on different floors it's very cumbersome to keep having to change each gridmap to the same floor I'm working on at that moment.

To clarify, this would be a setting to allow the floor for all gridmaps to be synched *optinally*.

**Comment by Nodragem** [comment]
oh yes, I see. Let's say you are using two Gridmap: `Gridmap1` and `Gridmap2`. You would like `Gridmap1`'s floor variable to change automatically to X  when you are working on `Gridmap2` and changes its floor variable to X. This way when you go back to working on `Gridmap1` you are already on the floor you want to edit.

To implement this, I believe we would need some sort of GridMap manager?

**Comment by elvisish** [comment]
> oh yes, I see. Let's say you are using two Gridmap: `Gridmap1` and `Gridmap2`. You would like `Gridmap1`'s floor variable to change automatically to X when you are working on `Gridmap2` and changes its floor variable to X. This way when you go back to working on `Gridmap1` you are already on the floor you want to edit.
> 
> To implement this, I believe we would need some sort of GridMap manager?

Yes, that might be an effective way of handling this. MeshLibraries *don't* share the same currently selected item, so when you switch between gridmaps, the mesh library item will crudely attempt to pick the same previous chosen index on every gridmap, which makes no sense if you have 2 mesh items in one gridmap and 70 in another, it just ends up switching back to 2 all the time, essentially: https://github.com/godotengine/godot/issues/89905

While this is a different issue entirely, the concept of a Gridmap Manager could be put to use for this as well (and perhaps however MeshLibraries keep the index seperate could be borrowed for Gridmap floors.
